pywemo/subscribe.py

```diff
@@ -241,8 +251,16 @@ class HTTPServer(ThreadingHTTPServer):
 
 def _start_server() -> HTTPServer | None:
     """Find a valid open port and start the HTTP server."""
-    for i in range(0, 128):
-        port = 8989 + i
+    requested_port = os.getenv('PYWEMO_HTTP_SERVER_PORT')
```

Comment:
> Please let me know if you think the constructor approach is more appropriate and I'm happy to go down that path.

Let's do both here. Pass the requested port in as an optional constructor arg and...

```python
def __init__(self, requested_port: int = 0) -> None:
    self.requested_port = requested_port
...


requested_port = int(os.getenv('PYWEMO_HTTP_SERVER_PORT', self.requested_port))
if requested_port:
...
```

I think it's okay to fail with a `ValueError` exception if the PYWEMO_HTTP_SERVER_PORT cannot be parsed as an `int`. That way there is some feedback/logging that what the user wanted had failed.

Reply:
> Let's do both here.

With the commit I just pushed, the precedence will be:

 * options passed to the constructor
 * environment variables
 * auto-detected values

This order makes sense to me.  If an application provides a way to configure these values, its probably better to take those values then value from the environment.  An argument could probably also be made to reverse the order, so I'm open to changing it as well.

Reply:
Sounds good to me. Thank you.

Reply:
Sorry for not being clearer initially. I had only intended to request that the `PYWEMO_HTTP_SERVER_PORT` could be passed in as a constructor arg. I was thinking `PYWEMO_CALLBACK_ADDRESS` would require more extensive changes.

Reply:
No worries, I don't mind iterating when I'm getting prompt feedback :).

It's a new codebase for me, so I was gathering context as I went.